worker-document-interface/src/main/java/com/hpe/caf/worker/document/extensibility/DocumentWorker.java

```diff
@@ -36,11 +36,21 @@ public interface DocumentWorker extends AutoCloseable
 {
     /**
      * This method provides an opportunity for the worker to report if it has any problems which would prevent it processing documents
-     * correctly. If the worker is healthy then it should simply return without calling the health monitor.
+     * correctly, and for which the worker might need to be restarted, for example, if the worker became deadlocked. If the worker is
+     * alive then it should simply return without calling the health monitor.
      *
-     * @param healthMonitor used to report the health of the application
+     * @param healthMonitor used to report the liveness of the application
      */
-    void checkHealth(HealthMonitor healthMonitor);
```

Comment:
- As discussed, use the existing `checkHealth()` for readiness (remember alive is just a subset) instead of introducing `checkReady()`.